x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/crossclusteraccess/CrossClusterAccessAuthenticationServiceIntegTests.java

```diff
@@ -135,16 +146,186 @@ public void testInvalidHeaders() throws IOException {
         }
     }
 
+    public void testTryAuthenticateCredentialsHeaderSuccess() throws IOException {
+        final String encodedCrossClusterAccessApiKey = getEncodedCrossClusterAccessApiKey();
+        final String nodeName = internalCluster().getRandomNodeName();
+        final ThreadContext threadContext = internalCluster().getInstance(SecurityContext.class, nodeName).getThreadContext();
+        final CrossClusterAccessAuthenticationService service = internalCluster().getInstance(
+            CrossClusterAccessAuthenticationService.class,
+            nodeName
+        );
+
+        try (var ignored = threadContext.stashContext()) {
+            addRandomizedHeaders(threadContext, encodedCrossClusterAccessApiKey);
```

Comment:
I could clean this up some more to only populate thread context headers once -- this might allow us to completely be rid of `doTryAuthenticateCredentialsHeader` and prevent passing in thread context at all. 

But I didn't want to block other work so leaving as is. Happy to revisit in a follow up.

Reply:
Why do we need to pass `threadContext` into the method `doTryAuthenticateCredentialsHeader`? The threadContext is a thread local. What you stash and prepare here will be exactly what the method sees from using `clusterService.threadPool().getThreadContext()`. Am I missing something?

Reply:
I rushed through this one too fast -- I could have sworn that this did not work with threadContext passed implicitly, but of course it does. Thanks!